src/main/java/com/rackspace/feeds/feedscatalog/ResolveHostFilter.java

```diff
@@ -0,0 +1,94 @@
+    static final String resolveHostXslt = "feedscatalog-resolve-host.xsl";
+
+    private TransformerUtils transformer;
+
+    /**
+     * This method is called once when the filter is first loaded.
+     */
+    public void init(FilterConfig filterConfig) throws ServletException {
+        try {
+            URL xsltResource = Thread.currentThread().getContextClassLoader().getResource(resolveHostXslt);
+            if (xsltResource != null) {
+                transformer = TransformerUtils.getInstanceForXsltAsFile(xsltResource.getPath());
+            }
+            else {
+                throw new UnavailableException("resource is null - " + resolveHostXslt);
+            }
+        } catch ( Exception e ) {
+            LOG.error( "Error loading xslt: " + e.getMessage());
+            throw new ServletException( e );
+        }
+    }
+
+    public void doFilter (ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+
+        HashMap<String, Object> params = new HashMap<String, Object>();
+        putTenantIdParams(httpServletRequest, params);
+        putExternalLocParams(httpServletRequest, params);
+
+        transformer.doTransform(httpServletRequest, httpServletResponse, httpServletResponse, chain, params);
+    }
+
+    /**
+     * The counterpart to the init( ) method.
+     */
+    public void destroy( ) {
+
+    }
+
+    void putTenantIdParams(HttpServletRequest request, HashMap<String, Object> params) {
+        //put tenantId and nastId into the params HashMap
+        //tenantId is the shortest length value, nastId is the longest length value
+        Enumeration headerValues = request.getHeaders("x-tenant-id");
+        ArrayList<String> tenantIds = new ArrayList<String>();
+        while (headerValues.hasMoreElements()) {
+            tenantIds.add((String) headerValues.nextElement());
+        }
+
+        if (tenantIds.size() > 0) {
+            String tenantId = tenantIds.get(0);
+            String nastId = tenantIds.get(0);
+
+            for (String s: tenantIds) {
+                if (s.length() < tenantId.length()) {
+                    tenantId = s;
+                }
+
+                if (s.length() > nastId.length()) {
+                    nastId = s;
+                }
+            }
+
+            params.put("tenantId", tenantId);
+            params.put("nastId", nastId);
+        }
+    }
+
+    void putExternalLocParams(HttpServletRequest request, HashMap<String, Object> params) {
+        //put x-external-loc header into the params HashMap if exists
+        String externalLocs = request.getHeader("x-external-loc");
+        if (externalLocs != null && !externalLocs.isEmpty()) {
+            params.put("externalLoc", externalLocs);
+        }
```

Comment:
Oh, I see you have a unit test that doesn't pass in the externalLoc parameter. If that works, then it's fine.